R/functions_netCDF.R

```diff
@@ -337,105 +401,238 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
-    dim = list(bnddim, xdim), missval = NULL,
-    chunksizes = c(2L, var_chunksizes[1]))
-  latbnddef <- ncdf4::ncvar_def(name = "lat_bnds", units = "",
-    dim = list(bnddim, ydim), missval = NULL,
-    chunksizes = c(2L, var_chunksizes[2]))
+  # define dimension bands ---------------------------------------------------------------
+  if(isGridded) {
+    lonbnddef <- ncdf4::ncvar_def(name = "lon_bnds", units = "",
+                                  dim = list(bnddim, xdim), missval = NULL,
+                                  chunksizes = c(2L, var_chunksizes[1]))
+
+    latbnddef <- ncdf4::ncvar_def(name = "lat_bnds", units = "",
+                                  dim = list(bnddim, ydim), missval = NULL,
+                                  chunksizes = c(2L, var_chunksizes[2]))
+  }
+
+  if (has_T_timeAxis) {
+      tbnddef <- ncdf4::ncvar_def(name = "time_bnds", units = "",
+                                dim = list(bnddim, tdim), missval = NULL,
+                                chunksizes = c(2L, 1L))
+  }
 
-  if (has_time_central) {
-    tbnddef <- ncdf4::ncvar_def(name = "time_bnds", units = "",
-      dim = list(bnddim, tdim), missval = NULL, chunksizes = c(2L, 1L))
+  if (has_Z_verticalAxis) {
+      vertbnddef <- ncdf4::ncvar_def(name = "depth_bnds", units = "",
+                                    dim = list(bnddim, zdim), missval = NULL,
+                                    chunksizes = c(2L, 1L))
   }
 
-  nc_dimvars <- list(lonbnddef, latbnddef)
-  if (has_time_central) {
+  nc_dimvars <- if(isGridded) list(lonbnddef, latbnddef) else list()
+
+  if (has_T_timeAxis) {
     nc_dimvars <- c(nc_dimvars, list(tbnddef))
   }
+  if (has_Z_verticalAxis) {
+    nc_dimvars <- c(nc_dimvars, list(vertbnddef))
+  }
+
+  # ---------------------------------------------------------------------------------------
+  #--- create empty netCDF file -----------------------------------------------------------
+  # ---------------------------------------------------------------------------------------
 
-  #--- create netCDF file
   dir.create(dirname(file), recursive = TRUE, showWarnings = FALSE)
+
   nc <- ncdf4::nc_create(filename = file,
     vars = c(nc_dimvars, list(crsdef), var_defs), force_v4 = force_v4)
+
   on.exit(ncdf4::nc_close(nc))
 
+ #--- write values of dimension bounds -----------------------------------------------------
+  if(isGridded) {
+    try(ncdf4::ncvar_put(nc, varid = "lon_bnds",
+                        vals = rbind(xvals - grid_halfres[1], xvals + grid_halfres[1]),
+                        start = c(1, 1), count = c(2L, var_chunksizes[1])))
+
+    try(ncdf4::ncvar_put(nc, varid = "lat_bnds",
+                         vals = rbind(yvals + grid_halfres[2], yvals - grid_halfres[2]),
+                         start = c(1, 1), count = c(2L, var_chunksizes[2])))
+  } else {
+    try(ncdf4::ncvar_put(nc, varid = "lon",
+                         vals = locations$X_WGS84,
```

Comment:
Earlier, we converted `loc <- sp::SpatialPoints(locations)`. Thus, we could use `sp::coordinates(locations)` instead of assuming a specific and undocumented naming of the xy-columns of `locations` - that also only make sense if the crs is WGS84

Reply:
fixed